module/VuFind/src/VuFind/RateLimiter/RateLimiterManager.php

```diff
@@ -171,6 +184,15 @@ public function check(EventInterface $event): array
                     ]
                 );
             }
+            if (isset($turnstileLimit)) {
+                $priorTurnstileResult = $this->checkPriorTurnstileResult($policyId, $this->clientIp);
+                if (!$priorTurnstileResult && !$turnstileLimit->isAccepted()) {
+                    $result['allow'] = false;
+                    $result['message'] = $this->getTooManyRequestsResponseMessage($event, $result);
```

Comment:
I don't think there's any need for a *different* message when Turnstile is the cause of the rate limit failure, but someone may disagree.